src/v/raft/consensus.cc

```diff
@@ -858,21 +858,20 @@ void consensus::dispatch_vote(bool leadership_transfer) {
                         return f;
                     }
                     // background
-                    (void)with_gate(
+                    ssx::spawn_with_gate(
                       _bg,
                       [vstm = std::move(vstm), f = std::move(f)]() mutable {
                           return std::move(f);
                       });
 
                     return ss::make_ready_future<>();
                 });
-          })
+          }).finally([this] { arm_vote_timeout(); });})
```

Comment:
Can you explain why this was moved?

Reply:
The finally is at the same scope as before (albeit reordered), but the handle_exception that came before it is moved to outside the spawn_with_gate_then.  The reason for that is to only hit the catch-all exception handler if the exception is not one of the ones that is caught silently by spawn_with_gate_then -- avoid logging warnings on gate closed etc.